pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -481,6 +486,30 @@ def _data_setter(
         with self.model:
             pm.set_data(data)
 
+    @classmethod
+    def _model_config_formatting(cls, model_config: Dict) -> Dict:
+        """
+        Because of json serialization, model_config values that were originally tuples or numpy are being encoded as lists.
+        This function converts them back to tuples and numpy arrays to ensure correct id encoding.
+        """
+        for key in model_config:
+            if isinstance(model_config[key], dict):
```

Comment:
This one is not recursive though, is it?

Reply:
oh, yes. That's not. I'll implement the same kind of function that is now for the lists